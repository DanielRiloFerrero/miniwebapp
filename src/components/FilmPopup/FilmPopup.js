import React, { useState, useEffect } from 'react'
import { Button, Modal, ModalHeader, ModalBody, ModalFooter, CardImg, FormGroup, Label, Input } from 'reactstrap';
import { format } from 'date-fns'

import { IMG_URL } from '../../_assets/js/const'
import parentalAd from '../../_assets/images/parental_advisory.svg'
import notAvailable from '../../_assets/images/not_available.svg'
import styles from './FilmPopup.module.scss'

const FilmPopup = props => {
  const [errorRate, setErrorRate] = useState('')
  const [errorTextarea, setErrorTextarea] = useState('')
  const [rateValue, setRateValue] = useState('')
  const [textareaValue, setTextareaValue] = useState('')

  const moveId = props.data.id
  const srcImg = props.data.backdrop_path ? `${IMG_URL}${props.data.backdrop_path}` : props.data.poster_path ? `${IMG_URL}${props.data.poster_path}` : notAvailable
  const originalTitle = props.data.original_title ? props.data.original_title : `Título no disponible`
  const overView = props.data.overview ? props.data.overview : `No disponible`
  const originalLanguage = props.data.original_language ? props.data.original_language : `Desconocido`
  const releaseDate = props.data.release_date ? format(new Date(props.data.release_date), 'dd/MM/yyyy') : 'Desconocida'
  const voteAverage = props.data.vote_average ? props.data.vote_average : null
  const parental = props.data.adult ? props.data.adult : null

  const movieRate = `movie_${moveId}`
  const movieTextarea = `textarea_${moveId}`

  useEffect(() => {
    if (sessionStorage[moveId]) {
      const movieValues = JSON.parse(sessionStorage[moveId])
      setRateValue(movieValues.rate)
      setTextareaValue(movieValues.textarea)
    }
  }, [moveId]);

  const handleSaveButton = () => {
    const myRate = document.querySelector(`#${movieRate}`).value
    const myTextarea = document.querySelector(`#${movieTextarea}`).value

    myRate === '' ? setErrorRate('La nota es obligatoria') : setErrorRate('')
    myTextarea === '' ? setErrorTextarea('El comentario es obligatorio') : setErrorRate('')

    if (myRate === '' || myTextarea === '') return false

    const info = {
      'rate': myRate,
      'textarea': myTextarea
    }

    sessionStorage[moveId] = JSON.stringify(info)
    props.close()
  }

  return (
    <div>
      <Modal isOpen={true} toggle={() => props.close()}>
        <ModalHeader>{originalTitle}</ModalHeader>

        <ModalBody>
          <div className={styles.imgWrapper}>
            <CardImg top width='100%' src={srcImg} alt='Card image cap' />
            {voteAverage && <div className={styles.average}>{voteAverage}</div>}
            {parental && <div className={styles.parental}><img src={parentalAd} alt='+18' /></div>}
          </div>

          <FormGroup><span className={styles.title}>Sinopsis: </span>{' '}<span>{overView}</span></FormGroup>
          <FormGroup><span className={styles.title}>Idioma original: </span>{' '}<span>{originalLanguage}</span></FormGroup>
          <FormGroup><span className={styles.title}>Fecha de esteno: </span>{' '}<span>{releaseDate}</span></FormGroup>

          <FormGroup>
            <Label className={styles.title} for={movieRate}>Mi puntuación:</Label>
            <Input value={rateValue} id={movieRate} type='number' onChange={e => setRateValue(e.target.value)} />
            {errorRate && <p>{errorRate}</p>}
          </FormGroup>
          <FormGroup>
            <Label className={styles.title} for={movieTextarea}>Crítica</Label>
            <Input value={textareaValue} type='textarea' name='text' id={movieTextarea} onChange={e => setTextareaValue(e.target.value)} />
            {errorTextarea && <p>{errorTextarea}</p>}
          </FormGroup>
        </ModalBody>

        <ModalFooter>
          <Button color='primary' onClick={() => handleSaveButton()}>Guardar</Button>{' '}
          <Button color='secondary' onClick={() => props.close()}>Cancel</Button>
        </ModalFooter>
      </Modal>
    </div>
  )
}

export default FilmPopup