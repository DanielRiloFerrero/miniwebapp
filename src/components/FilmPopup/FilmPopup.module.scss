@import '../../_assets/styles/abstracts/vars.scss';

.average {
  position: absolute;
  top: 0;
  right: 0;
  z-index: 1;
  background: #f00;
  padding: $p-xs;
  border-bottom-left-radius: 1rem;
  color: $secondary-color;
}

.imgWrapper {
  position: relative;
  margin-bottom: $m-s;
}

.parental {
  position: absolute;
    bottom: 0;
    left: 0;
    width: 5rem;
    display: block;

    img {
      display: block;
      max-width: 100%;
    }
}

.title {
  font-weight: 700;
}