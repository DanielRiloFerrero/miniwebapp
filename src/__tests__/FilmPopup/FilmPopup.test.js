import React from "react"
import { render, cleanup } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import FilmPopup from "../../components/FilmPopup/FilmPopup"

const data = {
  'id': 1,
  'backdrop_path': 'dummy',
  'original_title': 'dummy',
  'overview': 'dummy',
  'original_language': 'dummy',
  'release_date': new Date(),
  'vote_average': 9,
  'adult': false,
}

describe('FilmPopup tests', () => {
  it("render", () => {
    const { container } = render(<FilmPopup data={data} />);
    expect(container).toMatchSnapshot()
  })
})