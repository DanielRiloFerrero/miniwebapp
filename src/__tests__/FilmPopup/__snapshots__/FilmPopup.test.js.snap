// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmPopup tests render 1`] = `
<div>
  <div />
</div>
`;
